app/src/main/java/com/github/se/icebreakrr/MainActivity.kt

```diff
@@ -90,33 +93,51 @@ class MainActivity : ComponentActivity() {
  * @see NotificationScreen
  */
 @Composable
-fun IcebreakrrApp() {
-  val navController = rememberNavController()
-  val navigationActions = NavigationActions(navController)
+fun IcebreakrrApp(auth: FirebaseAuth?, functions: FirebaseFunctions?) {
   val profileViewModel: ProfilesViewModel = viewModel(factory = ProfilesViewModel.Factory)
   val tagsViewModel: TagsViewModel = viewModel(factory = TagsViewModel.Factory)
   val filterViewModel: FilterViewModel = viewModel(factory = FilterViewModel.Factory)
-  val ourUserUid = FirebaseAuth.getInstance().currentUser?.uid
-  val ourName = FirebaseAuth.getInstance().currentUser?.displayName
-  val functions = FirebaseFunctions.getInstance()
-  val meetingRequestViewModel: MeetingRequestViewModel =
+  val ourUserUid = auth?.currentUser?.uid ?: "null"
+  val ourName = auth?.currentUser?.displayName ?: "null"
+
+  MeetingRequestManager.meetingRequestViewModel =
       viewModel(
           factory =
-              MeetingRequestViewModel.Companion.Factory(
-                  profileViewModel, functions, ourUserUid, ourName))
+              functions?.let {
+                MeetingRequestViewModel.Companion.Factory(profileViewModel, it, ourUserUid, ourName)
+              })
+  val meetingRequestViewModel = MeetingRequestManager.meetingRequestViewModel
+
+  if (functions != null) {
+    IcebreakrrNavHost(
+        profileViewModel, tagsViewModel, filterViewModel, meetingRequestViewModel, Route.AUTH)
```

Comment:
Docs: Could you add more comments in that part as it is really hard for a "notification newbie" to understand what's done here